marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/IOEndpointImpl.java

```diff
@@ -274,5 +272,19 @@ public void interrupt() {
             if (this.phase == WorkPhase.RUNNING)
                 setPhase(WorkPhase.INTERRUPTING);
         }
+
+        void processOutput(InputStream[] output, Consumer<InputStream> outputConsumer) {
+            for (int i=allowsEndpointState()?1:0; i < output.length; i++) {
+                outputConsumer.accept(output[i]);
+            }
+        }
+
+        InputStream[] getOutput(InputStream[] output, Consumer<InputStream> outputConsumer) {
+            processOutput(output, outputConsumer);
+            if(allowsEndpointState()) {
```

Comment:
Should the if condition also make sure that the output.length > 0?

Arrays.copyOfRange(output, 1, 0) might throw an error.